src/components/ItemList/index.jsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState, useEffect } from 'react';
+import { updateItemDate } from '../Utils/firestore';
+import { isWithin24hours } from 'components/Utils/helpers';
+
+function ItemList({ itemName, docId, lastPurchasedDate }) {
+  const [isChecked, setIsChecked] = useState(false);
+  const [isDisabled, setIsDisabled] = useState(false);
+  const [token] = useState(() => window.localStorage.getItem('tcl18-token'));
+
+  useEffect(() => {
+    if (isWithin24hours(lastPurchasedDate)) {
+      setIsChecked(false);
+      setIsDisabled(false);
+    } else {
+      setIsChecked(true);
+      setIsDisabled(true);
+    }
+  }, [lastPurchasedDate]);
+
+  const MarkProductPurshased = (id) => {
```

Comment:
there's a small typo here, this might make it hard to find this function later now 😅 Purshased -> Purchased

Reply:
Fixed typo.